src/main/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncService.java

```diff
@@ -106,19 +104,15 @@ public void syncRecord(Record record) {
 
     TraineeDetailsDto dto = tableNameToMappingFunction.get(record.getTable()).apply(record);
 
-    if (record instanceof Person) {
-      if (hasRequiredRoleForProfileCreation(record)) {
-        personService.save((Person) record);
-      } else {
-        log.info("Trainee with id {} did not have the required role '{}'.", dto.getTraineeTisId(),
-            REQUIRED_ROLE);
-        return;
-      }
+    if (personService.findById(dto.getTraineeTisId()).isPresent()) {
+      Operation operationType = record.getOperation();
+      syncDetails(dto, apiPath.get(), operationType);
+    } else if (record instanceof Person && hasRequiredRoleForProfileCreation(record)) {
+      personService.save((Person) record);
```

Comment:
A question rather than anything else... do we also need to `syncDetails(dto, apiPath.get(), operationType);` or is that already handled by db event listeners?